server/lib/chat/chat-service.ts

```diff
@@ -443,14 +475,31 @@ export default class ChatService {
     channelName: string,
     userId: SbUserId,
   ): Promise<SbUserId | null> {
-    const { newOwnerId } = await leaveChannel(userId, channelName)
+    const { newOwnerId } = await removeUserFromChannel(userId, channelName)
+
+    // Notify the new owner that their permissions changed, but only if they're online.
```

Comment:
I don't really see the point in checking if they're online, we have a consistent route structure to make it easy to dispatch messages to particular users, just dispatch the message to the route. If they're not online, no sockets will get the message, done. The whole nice part about nydus over e.g. socket.io is this idea of persistent route subscriptions, if you go around checking if sockets are registered every time it sort of defeats the purpose

Reply:
I guess that comment is misleading; my idea with checking if they're online was just as a slight optimization to avoid making a database query fetching their permissions if they're not. If you think the optimization is not worth it, I can just remove it.

Reply:
That optimization is not worth it

Reply:
If the query above is already returning an owner ID, it could likely just return the new permissions as well, if you really want to optimize that. (But generally, this is a query for 1 row from a single table using only the primary key of the table, it's not really every going to be an expensive thing to do)